docs/advanced-topics/client-anticipation.md

```diff
@@ -0,0 +1,117 @@
+</figure>
+
+With client anticipation, we change the way this works: client 1 anticipates the change to blue so it happens immediately, and then later sees the object change to red, which, again, is fine. Client 2 also sees the object change to red immediately, but because a change to blue is already in progress, that overwrites client 2's anticipated value, causing it to flicker briefly to blue from client 1's request before changing back to red again from client 2's request.
+
+<figure>
+<ImageSwitcher 
+lightImageSrc="/sequence_diagrams/Anticipation/StaleDataNoPolicy.png?text=LightMode"
+darkImageSrc="/sequence_diagrams/Anticipation/StaleDataNoPolicy_Dark.png?text=DarkMode"/>
+</figure>
+
+To address this, Netcode for GameObjects's client anticipation includes a feature called `StaleDataHandling`. Stale data is determined based on assumptions about causation - it assumes that when you make an anticipation on the client side based on player input, that at the same time, you will be sending an RPC to the server requesting it to make the same change. It uses a continuously incrementing `AnticipationCounter` to track when the server has received and responded to the batch of requests that was sent on the same frame as the variable was anticipated. If an update for a variable arrives before the server has processed that message, the anticipation system regards that data as being "stale".
+
+There are two ways you can respond to stale data, which are determined by the `StaleDataHandling` value on each `AnticipatedNetworkVariable` and `AnticipatedNetworkTransform`:
+
+If `StaleDataHandling` is set to `StaleDataHandling.Ignore`, stale data will not roll back the value of the variable or transform to the server value and will not trigger the `OnReanticipate` event. `ShouldReanticipate` will remain false in the event something else triggers the callback. The authoritative value will still be updated, however, and for `AnticipatedNetworkVariable`, the `OnAuthoritativeValueUpdated` callback will still be called. The result here is that, for client 2, the change to blue is recognized as being sequenced before its change to red, and is thus ignored, eliminating the flickering. This is the default behavior for `AnticipatedNetworkVariable<T>`.
+
+<figure>
+<ImageSwitcher 
+lightImageSrc="/sequence_diagrams/Anticipation/StaleDataIgnore.png?text=LightMode"
+darkImageSrc="/sequence_diagrams/Anticipation/StaleDataIgnore_Dark.png?text=DarkMode"/>
+</figure>
+
+If `StaleDataHandling` is set to `StaleDataHandling.Reanticipate`, stale data will be treated the same way as any other server data updates. The value will be rolled back, `ShouldReanticipate` will be set to true, and the `OnReanticipate` event will be fired (see below). In typical client prediction systems, and generally involves replaying the player's input from the time of the incoming data to now, which results in re-performing the switch to red.
+
```

Comment:
@jabbacakes
Not anything to hold this PR up, but we should see if the samples team has any bandwidth to create a series of companion examples to this documentation as it could be confusing to try to replicate the code required for this. Kitty already provided a generlized sample that has a lot of this done, so it might not be too time consuming to make something (based on the sample Kitty provided) that would walk users through this entire page of documentation.